src/containers/CreatePipelineRun/YAMLEditor.js

```diff
@@ -23,21 +27,97 @@ import React, { useState } from 'react';
 import CodeMirror from '@uiw/react-codemirror';
 import { StreamLanguage } from '@codemirror/language';
 import { yaml as codeMirrorYAML } from '@codemirror/legacy-modes/mode/yaml';
-import { useNavigate } from 'react-router-dom-v5-compat';
-import { createPipelineRunRaw, useSelectedNamespace } from '../../api';
+import { useLocation, useNavigate } from 'react-router-dom-v5-compat';
+import deepClone from 'lodash.clonedeep';
+import {
+  createPipelineRunRaw,
+  usePipelineRun,
+  useSelectedNamespace
+} from '../../api';
+
+export function generateNewPipelineRun(pipelineRun) {
```

Comment:
Could we reuse the existing code from the `rerunPipelineRun` API? Similar to what we've done with `createPipelineRun`, having a function to generate the sanitised payload.

Reply:
what about this one? we are not kinda re-running
```
'dashboard.tekton.dev/rerunOf': name
```
why do we need this one?
```
delete payload.metadata.labels['tekton.dev/pipeline'];
```

Reply:
`'dashboard.tekton.dev/rerunOf': name`

Yes, we can skip that one 👍 

`delete payload.metadata.labels['tekton.dev/pipeline'];`

We shouldn't set any `tekton.dev/*` labels, they're reserved for use by the Tekton Pipelines controllers, and adding them on the input could potentially cause issues. See https://github.com/tektoncd/pipeline/issues/4366 for some context.